pkg/3scale/amp/operator/upgrade.go

```diff
@@ -530,6 +530,10 @@ func (u *UpgradeApiManager) ensureSystemAppMonitoringSettings() (bool, error) {
 }
 
 func (u *UpgradeApiManager) upgradePrometheusRules(desired *monitoringv1.PrometheusRule) (bool, error) {
+	if !u.apiManager.IsMonitoringEnabled() {
+		return false, nil
+	}
```

Comment:
@eguzki apologies if this seems obvious, but seeing as I'm not at all acquainted with the codebase here: 

i see this changes the behaviour of `upgradePrometheusRules` which I presume crashed with some error when monitoring was disabled. 

am I right in understanding the bug didn't allow control flow to reach [here](http://github.com/3scale/3scale-operator/blob/5b54f6adffc48785d35d444649ac1169d0dc26c8/pkg/3scale/amp/operator/upgrade.go#L442-L442), because of some error [here](http://github.com/3scale/3scale-operator/blob/5b54f6adffc48785d35d444649ac1169d0dc26c8/pkg/3scale/amp/operator/upgrade.go#L410-L410) ?

Reply:
> am I right in understanding the bug didn't allow control flow to reach here, because of some error here ?
Yes. That is what happened before the fix. The root issue is that [here](https://github.com/3scale/3scale-operator/blob/5b54f6adffc48785d35d444649ac1169d0dc26c8/pkg/3scale/amp/operator/upgrade.go#L538), the operator tries to read a k8s resource that expects to exist to upgrade. The operation fails when the object does not exist. That expectation is wrong when monitoring is disabled.

Reply:
ok, thanks, yes. 👍